packages/react-log-viewer/src/LogViewer/LogViewer.tsx

```diff
@@ -0,0 +1,157 @@
+    hasToolbar = true,
+    customControls = null,
+    placeholder = 'Search',
+    hasLineNumbers = true,
+    width = 800
+  }: LogViewerProps) => {
+    const [searchedInput, setSearchedInput] = useState<string | null>('');
+    const [rowInFocus, setRowInFocus] = useState<number | null>(null);
+    const [searchedWordIndexes, setSearchedWordIndexes] = useState<number[] | null>([]);
+    const [highlightedRowIndexes, setHighlightedRowIndexes] = useState<number[] | null>([]);
+    const [currentSearchedItemCount, setCurrentSearchedItemCount] = useState<number>(0);
+    const [parsedData, setParsedData] = useState<string[] | null>([]);
+    const loggerRef = React.createRef<any>();
+
+    const dataToRender = React.useMemo(
+      () => ({
+        parsedData,
+        loggerRef,
+        rowInFocus,
+        searchedWordIndexes,
+        highlightedRowIndexes,
+        setHighlightedRowIndexes
+      }),
+      [parsedData, loggerRef, rowInFocus, searchedWordIndexes, highlightedRowIndexes, setHighlightedRowIndexes]
+    );
+
+    /* Parsing our data immediately after first component render */
+    useEffect(() => {
+      setParsedData(parseConsoleOutput(data));
+    }, [data]);
+
+    /* Updating searchedResults context state given changes in searched input */
+    useEffect(() => {
+      let foundKeywordIndexes: (number | null)[] = [];
+      const adjustedSearchedInput = escapeString(searchedInput);
+
+      if (adjustedSearchedInput !== '' && adjustedSearchedInput.length >= 3) {
+        foundKeywordIndexes = searchForKeyword(adjustedSearchedInput, parsedData);
+
+        if (foundKeywordIndexes.length !== 0) {
+          setSearchedWordIndexes(foundKeywordIndexes);
+          scrollToRow(foundKeywordIndexes[DEFAULT_SEARCH_INDEX]);
+        }
+      }
+
+      if (adjustedSearchedInput !== '' && adjustedSearchedInput.length < 3) {
+        setRowInFocus(DEFAULT_FOCUS);
+        setCurrentSearchedItemCount(DEFAULT_INDEX);
+        setSearchedWordIndexes([]);
+      }
+
+      if (adjustedSearchedInput === '') {
+        setRowInFocus(DEFAULT_FOCUS);
+        // setCurrentSearchedItemCount(DEFAULT_INDEX);
+        // setSearchedWordIndexes([]);
+      }
+    }, [searchedInput]);
+
+    const scrollToRow = (searchedRowIndex: number) => {
+      setRowInFocus(searchedRowIndex);
+      loggerRef.current.scrollToItem(searchedRowIndex, 'center');
+      return true;
+    };
+
+    const loggerWidth = loggerRef.current ? loggerRef.current.clientWidth : 800;
+    const guessRowHeight = (rowIndex: number) => {
+      const rowText = parsedData[rowIndex];
+      const textWidth = getTextWidth(rowText, 'Liberation Mono');
+      const numRows = Math.ceil(textWidth / loggerWidth);
+
+      return 60 * numRows;
+    };
+
+    return (
+      <LogViewerContext.Provider
+        value={{
+          parsedData
+        }}
+      >
+        <div className={css(styles.logViewer) + (hasLineNumbers ? ' pf-m-line-numbers' : '')}>
+          <LogViewerToolbar
```

Comment:
Should be wrapped in `hasToolbar` still. Or is this to change later on in a followup?